provider/etcd.go

```diff
@@ -8,12 +10,16 @@ import (
 
 // Etcd holds configurations of the Etcd provider.
 type Etcd struct {
-	Kv `mapstructure:",squash"`
+	Kv  `mapstructure:",squash"`
+	TLS *KvTLS
```

Comment:
You already added `TLS *KvTLS` in the `Kv` struct. You should remove this one ;)